R/deploymentTarget.R

```diff
@@ -1,93 +1,241 @@
 # calculate the deployment target based on the passed parameters and
 # any saved deployments that we have
-deploymentTarget <- function(recordPath = ".",
-                             appId = NULL,
-                             appName = NULL,
-                             appTitle = NULL,
-                             envVars = NULL,
-                             account = NULL,
-                             server = NULL,
-                             forceUpdate = FALSE,
-                             error_call = caller_env()) {
+deploymentTarget <- function(
+  recordPath = ".",
+  appId = NULL,
+  appName = NULL,
+  appTitle = NULL,
+  envVars = NULL,
+  account = NULL,
+  server = NULL,
+  forceUpdate = FALSE,
+  error_call = caller_env()
+) {
 
-  appDeployments <- deployments(
+  if (!is.null(appId)) {
+    return(deploymentTargetFromAppId(
+      recordPath = recordPath,
+      appId = appId,
+      appName = appName,
+      appTitle = appTitle,
+      envVars = envVars,
+      account = account,
+      server = server,
+      error_call = error_call
+    ))
+  }
+
+  if (!is.null(appName)) {
```

Comment:
```suggestion
  } else if (!is.null(appName)) {
```

? Too make it more clear we're grouping two early returns together.

Reply:
My initial feeling is to keep this as-is; most of these deployment discovery functions are using early returns. Shifting to an if-else makes me want to group the remainder of the function in a massive "else" clause, which leads to more indentation.

It's a minor style preference.

Reply:
:+1: